src/utils/hooks/useKubevirtDataPod/hooks/useKubevirtDataPodHealth.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useRef } from 'react';
+
+import { consoleFetch } from '@openshift-console/dynamic-plugin-sdk';
+
+import { PROXY_KUBEVIRT_URL, PROXY_KUBEVIRT_URL_HEALTH_PATH } from '../utils/constants';
+
+const useKubevirtDataPodHealth = () => {
+  const alive = useRef(null);
```

Comment:
Don't we need a `useState` here?
This still works because the  `useKubevirtWatchResource` trigger the re-render several times and in the meantime, this `consoleFetch` resolve before the `useWatchResource` in `useKubevirtWatchResource` could end fetching data. Is there a case where we can have problems?

Reply:
u mean ref ? 
Because of the ref, it will trigger /health call only once for each load of hook and won't be affected by rerenders. on line 19 I'm checking before calling that ref has no results, ref will be declared and used on all lifecycles of calling component with the same result and isn't affected by rerenders

Reply:
Cool!!